main.tf

```diff
@@ -43,7 +43,7 @@ resource "aws_api_gateway_deployment" "this" {
   ]
 
   rest_api_id = "${aws_api_gateway_rest_api.this.id}"
-  stage_name  = ""
+  stage_name  = "${var.deployment_stage}"
```

Comment:
@erikbor is this safe? the `stage_name` is set to be empty to avoid AWS  generate default CloudWatchLog. See https://github.com/terraform-providers/terraform-provider-aws/issues/2406#issuecomment-392666182